### PR TITLE
skeleton for custom domains documentation, missing REST changes

### PR DESCRIPTION
**Pull request by macolso**
Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title, `template`, and `date` are all set
- [ ] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)

a) Yes! This PR has new menu items; therefore, we need to ensure that corresponding pages (that the menu links to) exist before we approve and merge.
b) This PR creates new pages and therefore we need to be sure that all pages that exist have a corresponding menu item. So there are no pages buried without a direct link from the menu. (You can run [this script](https://github.com/fermyon/developer/issues/646#issuecomment-1553981354) if you prefer a machine to do the page:menu validation)

- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: spin.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ trigger = { type = "http", base = "/" }
 [[component]]
 source = "modules/bartholomew.wasm"
 id = "bartholomew"
-environment = { PREVIEW_MODE = "0" }
+environment = { PREVIEW_MODE = "1" }
```
Comment (tpmccallum) [inline spin.toml:11]:
```suggestion
environment = { PREVIEW_MODE = "0" }
```

**Review by tpmccallum — COMMENTED** [review]
Just setting `PREVIEW_MODE` back to zero. 
My understanding is that we leave this at zero all of the time and if we want to run the developer documentation locally to test we use the `spin up -e "PREVIEW_MODE=1"` command in our local CLI.
From memory, changing this value in GH makes the files go live instantly (instead of obeying the frontmatter).

**Comment by tpmccallum** [comment]
Does anyone know why the DCO is failing on this PR?

**Comment by itowlson** [comment]
@tpmccallum it means one of the commits is not signed. Pointing no fingers but it's totally @macolso

**Comment by tpmccallum** [comment]
Ah, so it puts the red X on the whole PR if only one commit is not signed. Got it.

**Comment by tpmccallum** [comment]
Just for convenience, @macolso there is a bit about signing in the following contributing to docs page: https://developer.fermyon.com/spin/contributing-docs#9-commit-changes

If committing in the CLI, the trick is to use both `-S` and `--signoff`
If committing in the UI, the trick is to format the signing message as follows and make that message the first part of the commit comment.

`Signed-off-by: username email`

For example, use the output from the following CLI commands to populate the signing message in the UI.

```console
tpmccallum@192-168-1-14 hello-docker % git config user.name
tpmccallum
tpmccallum@192-168-1-14 hello-docker % git config user.email
tim.mccallum@fermyon.com
```

`Signed-off-by: tpmccallum tim.mccallum@fermyon.com`

Hope this helps, please let me know if you have any questions.

**Comment by macolso** [comment]
Whoops - sorry team! I glossed over that step by accident. Thanks for the refresher. I'll close this PR and open a new one.

**Comment by macolso** [comment]
See https://github.com/fermyon/developer/pull/739

**Pull request closed without merging**